pymeasure/instruments/oxfordinstruments/ips120_10.py

```diff
@@ -0,0 +1,330 @@
+    _MAX_VOLTAGE = 10  # Volts
+    _SWITCH_HEATER_DELAY = 20  # Seconds
+
+    _TRAINING_SCHEME = [  # (field, field-rate)-pairs
+        (11.8, 1.0),
+        (13.9, 0.4),
+        (14.9, 0.2),
+        (16.0, 0.1),
+    ]
+
+    def __init__(self, resourceName, clear_buffer=True, switch_heater_delay=None, **kwargs):
+        super().__init__(
+            resourceName,
+            "Oxford IPS",
+            includeSCPI=False,
+            send_end=True,
+            read_termination="\r",
+            **kwargs
+        )
+
+        if switch_heater_delay is not None:
+            self._SWITCH_HEATER_DELAY = switch_heater_delay
+
+        # Clear the buffer in order to prevent communication problems
+        if clear_buffer:
+            self.adapter.connection.clear()
+
+    version = Instrument.measurement(
+        "V",
+        """ A string property that returns the version of the IPS. """
+    )
+
+    control_mode = Instrument.control(
+        "X", "$C%d",
+        """ A string property that sets the IPS in LOCAL or REMOTE and LOCKEd,
+        or UNLOCKED, the LOC/REM button. Allowed values are:
+        LL: LOCAL & LOCKED
+        RL: REMOTE & LOCKED
+        LU: LOCAL & UNLOCKED
+        RU: REMOTE & UNLOCKED. """,
+        get_process=lambda v: int(v[6]),
+        validator=strict_discrete_set,
+        values={"LL": 0, "RL": 1, "LU": 2, "RU": 3},
+        map_values=True,
+    )
+
+    current_measured = Instrument.measurement(
+        "R1",
+        """ A floating point property that returns the measured magnet current of
+        the IPS in amps. """,
+        get_process=lambda v: float(v[1:]),
+    )
+
+    demand_current = Instrument.measurement(
+        "R0",
+        """ A floating point property that returns the demand magnet current of
+        the IPS in amps. """,
+        get_process=lambda v: float(v[1:]),
+    )
+
+    demand_field = Instrument.measurement(
+        "R7",
+        """ A floating point property that returns the demand magnetic field of
+        the IPS in Tesla. """,
+        get_process=lambda v: float(v[1:]),
+    )
+
+    persistent_field = Instrument.measurement(
+        "R18",
+        """ A floating point property that returns the persistent magnetic field of
+        the IPS in Tesla. """,
+        get_process=lambda v: float(v[1:]),
+    )
+
+    field_polarity = Instrument.measurement(
+        "X",
+        """ A integer property that returns the magnet polarity of
+        the IPS in Tesla. """,
+        get_process=lambda v: int(v[13]),
+    )
+
+    polarity_contactor = Instrument.measurement(
+        "X",
+        """ A integer property that returns the contactor status of
+        the IPS in Tesla. """,
+        get_process=lambda v: int(v[14]),
+    )
+
+    switch_heater_status = Instrument.control(
+        "X", "$H%d",
+        """ A integer property that returns the switch heater status of
+        the IPS in Tesla. """,
+        get_process=lambda v: int(v[8]),
```

Comment:
do these ints also map to human-understandable strings, like for sweep_status?

Reply:
Yes they do; unfortunately, they're not as symmetric though: the value you set to the system do however not mean the same as the value you get from the system. I will try to make a property manually to get it a bit neater

Reply:
I've added an additional boolean property (`switch_heater_enabled`) that wraps around the `switch_heater_status` to take care of this.